src/metatensor/models/utils/data/dataset.py

```diff
@@ -15,20 +13,26 @@
 logger = logging.getLogger(__name__)
 
 
-if os.environ.get("METATENSOR_IMPORT_FOR_SPHINX", "0") == "1":
-    # This is necessary to make the Sphinx documentation build
-    def compiled_slice(a, b):
-        pass
+@dataclass
+class DatasetInfo:
+    """A class that contains information about one or more datasets.
 
-    def compiled_join(a, axis, remove_tensor_name):
-        pass
+    This dataclass is used to communicate additional dataset details to the
+    training functions of the individual models.
 
-else:
-    compiled_slice = torch.jit.script(metatensor.torch.slice)
-    compiled_join = torch.jit.script(metatensor.torch.join)
+    :param length_unit: The unit of length used in the dataset.
+    :param outputs: The names of the outputs in the dataset.
+    :param output_quantities: The quantities of the outputs in the dataset.
+    :param output_units: The units of the outputs in the dataset.
+    """
+
+    length_unit: str
+    outputs: List[str]
+    output_quantities: Dict[str, str]
+    output_units: Dict[str, str]
```

Comment:
I don't love that the output key is duplicated three times =/

Maybe `outputs = Dict[str, OutputInfo]`, with

```
@dataclass
class OutputInfo:
    quantity: str
    unit: str,
```


I would also call these `targets`, not `outputs`.

Reply:
I knowww but people will be scared off by all these nested custom classes

Reply:
Cool for target instead of output

Reply:
> I knowww but people will be scared off by all these nested custom classes

Will they though? We can also do namedtuple or even dict (with validation of the keys), i.e. `outputs: Dict[str, Dict[str, str]]`

Reply:
Fine, but I'm very concerned

Reply:
We can have a chat about this in tomorrow meeting to get more points of view!